Sage/carbon
# docs(number): fix bug where the input would be cleared on typing a letter - FE-5105

On the default example in storybook, if you type a letter, the whole input clears.

fixes FE-5105

### Proposed behaviour

![number_input_bug_fixed](https://user-images.githubusercontent.com/56251247/160863306-3f350ebe-04ba-4e95-bd5c-ccf4d98af7fc.gif)

### Current behaviour

![number_input_bug](https://user-images.githubusercontent.com/56251247/160863361-c49f32bb-6a6b-47a9-81a7-d3edc00af03d.gif)

### Checklist




#### QA


### Additional context



### Testing instructions

- Visit the `default` example in the Storybook for `Number Input`
- Type in `1234f` and the input should not clear it's value. Value in the input field should be `1234`
